classes/mail/MailTemplate.inc.php

```diff
@@ -149,7 +149,7 @@ function assignParams($params = array()) {
 		if ($this->context) {
 			// Add context-specific variables
 			$router = $request->getRouter();
-			$dispatcher = $request->getDispatcher();
+			$dispatcher = $application->getDispatcher();
```

Comment:
This should fix it; mind cherry-picking this into your branch to include it in the PR?
https://github.com/asmecher/pkp-lib/commit/0bd6ecea69c99002ec68f6c51fa11098cdec7ee3